source/material_model/rheology/frank_kamenetskii.cc

```diff
@@ -0,0 +1,118 @@
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/rheology/frank_kamenetskii.h>
+#include <aspect/utilities.h>
+
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/base/parameter_handler.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      template <int dim>
+      FrankKamenetskii<dim>::FrankKamenetskii ()
+      {}
+
+
+
+      template <int dim>
+      double
+      FrankKamenetskii<dim>::compute_viscosity (const double reference_viscosity,
+                                                const double temperature,
+                                                const unsigned int composition) const
+      {
+        // Frank-Kamenetskii viscosity approximation
+        //    viscosity = ref_v * A * B * exp(E * (0.5-(T/ref_T)/2)))
+        // ref_v: reference viscosity,
+        // A: prefactor, B: pre-limiter, E: adjusted viscosity ratio,
+        // ref_T: reference temperature, T: temperature.
+
+        const double viscosity_frank_kamenetskii = reference_viscosity * prefactors_frank_kamenetskii[composition]
+                                                   * std::exp(viscosity_ratios_frank_kamenetskii[composition]*(0.5-temperature/reference_T/2));
+
+        return viscosity_frank_kamenetskii;
+      }
+
+
+
+
+
+      template <int dim>
+      void
+      FrankKamenetskii<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.declare_entry ("Viscosity ratios Frank Kamenetskii", "15",
+                           Patterns::List(Patterns::Double (0.)),
+                           "An adjusted viscosity ratio, $E$, for the viscosity approximation, "
+                           "for a total of N+1 values, where N is the number of compositional fields. "
+                           "If only one value is given, then all use the same value. "
+                           "Units: None");
+        prm.declare_entry ("Prefactors Frank Kamenetskii", "1",
+                           Patterns::List(Patterns::Double (0.)),
+                           "A viscosity scaling prefactor for the viscosity approximation, "
+                           "for a total of N+1 values, where N is the number of compositional fields. "
+                           "If only one value is given, then all use the same value.  Units: None");
+        prm.declare_entry ("Reference temperature", "293", Patterns::Double(0),
+                           "For calculating density by thermal expansivity. Units: $K$");
+      }
+
+
+
+      template <int dim>
+      void
+      FrankKamenetskii<dim>::parse_parameters (ParameterHandler &prm)
+      {
+        // increment by one for background:
+        const unsigned int n_fields = this->n_compositional_fields() + 1;
+
```

Comment:
Since we discussed that for the moment we want this to be limited to models without an adiabatic temperature gradient, we should make sure that adiabatic heating is switched off. This can be done here the following way:
```
AssertThrow (this->include_adiabatic_heating() == false,
             ExcMessage("The Frank-Kamenetskii rheology is currently only implemented for "
                        "models without adiabatic heating. Please implement the necessary "
                        "temperature adjustment if you need this feature."));
```